### ISSUE TITLE
Old docs still showing up as top google results

### ISSUE THREAD

**Issue by johanandren** [akka/akka#24101]
As pointed out bu @sourcekick, for example `akka streams cookbook` still shows some old docs as top results, is this something we missed when we fixed this before?

https://gitter.im/akka/akka?at=5a259c36c65707ba2b0faaed

Ping @raboof

**Comment by raboof** [comment]
Hmm, `akka streams cookbook` does not point to `akka-stream-and-http-experimental` for me, but to https://doc.akka.io/docs/akka/2.5.3/scala/stream/stream-cookbook.html which isn't great either.

The main approach to fix this is using canonical links. For this page unfortunately the link in the page itself is invalid, but the link in the `Link` header correctly points to the `current` version. I wonder if google got confused by that, but I don't think it should.

I'll 'manually' remove https://doc.akka.io/docs/akka/2.5.3/scala/stream/stream-cookbook.html from the google cache, then let's see if it correctly picks up the canonical link header in a week or so.

**Comment by raboof** [comment]
I'm now getting https://doc.akka.io/docs/akka/2.5.5/scala/stream/stream-cookbook.html and ttps://doc.akka.io/docs/akka/2.5-M2/scala/stream/stream-cookbook.html . The canonical link was not *quite* right (contained a spurious `/`), so fixed that. Removing those 2 from the google cache.

tldr: after this has propagated we should have proper 'canonical' links for 2.5.3, 2.5.5 and 2.5-M2, so we shouldn't see those in the google results yet. If we're indeed convinced the multiple-redirects problem above was the reason google indexed the old URL's, we can mass-remove old URL's from Google's cache. If, on the other hand, 2.5.3, 2.5.5 or 2.5-M2 pop up in search results again later, the canonical headers may not work as we expect them to...

**Comment by raboof** [comment]
so far so good, 'current' is now the top result, now removed 2.4.19 from the index as well

**Comment by raboof** [comment]
also removed:
* https://doc.akka.io/docs/akka/2.4/scala/stream/stream-cookbook.html
* https://doc.akka.io/docs/akka/2.5.1/java/stream/stream-cookbook.html

I did realize just now that the fact that we don't see the old pages pop up again doesn't tell us anything yet, because they're still ('forcefully') 'temporarily hidden' - so this doesn't tell us what would happen if Google would index those pages again... I guess we don't want to wait those 90 days and should just start blocking outdated results en masse, though it is a bit 'flying blind' :/

**Comment by raboof** [comment]
Ok, so, today, https://doc.akka.io/docs/akka/2.4/scala/stream/stream-cookbook.html is back in the search results :(.

This suggests our 'canonical' links in HTTP headers aren't working so great after all. Perhaps because the page *also* contains a 'canonical' HTML, but that link points to a HTTP URL that produces a 301?

**Comment by raboof** [comment]
Further investigation suggests that 'canonical' links in HTTP headers are only interpreted for resources other than text/html pages (e.g. PDF's). Additionally, 'canonical' links that lead to 301 redirects are ignored. This means we should update the canonical links to point to the correct places.
